Airassault capture trigger

RndName opened the pull request:
Adds a capture trigger zone and a corresponding trigger rule for AirAssault target ControlPoints whenever the TriggerGenerator would not automatically generate them (which is only done for Fobs at the moment).

I've choosen that approach instead of just adding Airfield to capture_zone_types to prevent adding more trigger rules then needed which may cost performance (dont know about that but well its DCS..)

fixes #2566

DanAlbert commented:
> I've choosen that approach instead of just adding Airfield to capture_zone_types to prevent adding more trigger rules then needed which may cost performance (dont know about that but well its DCS..)

Actually I think I'd prefer that for consistency. I'm skeptical that the performance cost will be measurable compared to the rest of the mission? If a linear scaling of the number of capturable zones has a significant impact on mission performance we have other problems, because FOBs see pretty heavy use already.

If it is a real problem, I think this PR is generating a trigger for each air assault flight rather than per-target, so it'll potentially perform worse anyway?

RndName commented:
Well if thats the preferred way. Should we then ignore the dcs capture event?

And tbh air assault will be a rare tasking so thats for sure way less Triggers dont get that argument but im fine with just adding all airfields

DanAlbert commented:
> Well if thats the preferred way. Should we then ignore the dcs capture event?

SGTM

RndName committed:
Add Airfield to list of capture zone types

This will create capture zones and the trigger rules to check for a base capture. Will fix an issue where the dcs capture event is not fired and therefore the capture not recognized by liberation

RndName committed:
Remove dcs capture event from state json

With the latest change we added capture zones and corresponding trigger rules for all Airfields as well so we do not need to rely on the dcs capture event S_EVENT_BASE_CAPTURED anymore.

RndName commented:
@DanAlbert i did some testing with just using the custom capture zone trigger rule and it worked as expected but i am unsure if we should already put this into the planned release as there may be some corner cases regarding the capture itself. Otherwise we already rely on that trigger rule because of the Fobs so it is already working and tested. But we never tested it on an acutal Airfield which can have way more stuff around as the Fob...
What do you think about that one?

DanAlbert commented:
I think go ahead and merge. Like you say, this code is already working elsewhere.
